lib/ExpensiMark.js

```diff
@@ -250,4 +254,98 @@ export default class ExpensiMark {
 
         return replacedText;
     }
+
+    /**
+     * Modify text for Quotes replacing chevrons with html elements
+     *
+     * @param {RegExp} regex
+     * @param {String} textToCheck
+     *
+     * @returns {String}
+     */
+
+    modifyTextForQuote(regex, textToCheck) {
+        let replacedText = '';
+        let TextToFormat = '';
+        const codefenceSplitted = [];
+        const match = textToCheck.match(regex);
+        const match_codefence = textToCheck.match(/<pre>(\s|\S)+<\/pre>/gm);
+
+        // if there's matches we need to modify the quotes
+        if (match !== null) {
+            // split all the codefences in lines
+            for (let i = 0; i < match_codefence.length; i++) {
+                codefenceSplitted[i] = match_codefence[i].split('\n');
+            }
+
+            // split the textToCheck in lines
+            const textSplitted = textToCheck.split('\n');
+
+            for (let i = 0; i < textSplitted.length; i++) {
+                let isCodeFence = false;
+                for (let j = 0; j < codefenceSplitted.length; j++) {
+                    for (let k = 0; k < codefenceSplitted[j].length; k++) {
+                    // we need to know when is codefenced
+                        if (textSplitted[i] === codefenceSplitted[j][k]) {
+                            isCodeFence = true;
+                        }
+                    }
+                }
+
+                // we only want to modify lines starting with &gt that is not codefence;
+                if (textSplitted[i].substr(0, 4) === '&gt;' && !isCodeFence) {
+                    // avoid blank lines starting with &gt;
+                    if (textSplitted[i].trim() !== '&gt;') {
+                        textSplitted[i] = textSplitted[i].substr(4, textSplitted[i].length);
```

Comment:
I think if you're trying to get a substring from a certain index to the end of the string, the second parameter is redundant:

```suggestion
                        textSplitted[i] = textSplitted[i].substr(4).trim();
```

Reply:
Ok